core/src/Repository/DataSource/Sql/Helper/SqlBuilder.php

```diff
@@ -138,12 +134,47 @@ public function multiInsert(array $entities): Query {
     return $query;
   }
 
-  public function deleteById(mixed $value): Query {
-    $query = $this->factory
-      ->delete($this->schema->getTableName())
-      ->where(field($this->schema->getIdColumn())->eq($value))
-      ->compile();
+  public function deleteById(mixed $value): SqlQuery {
+    if ($this->schema->softDeleteEnabled()) {
+      $now = date("Y-m-d H:i:s");
```

Comment:
@orioljp please, ping me when you have time, to discuss the column type for delete_at.